Error opening a Vulkan trace from SaschaWillems demo: The gapis server has exited with an error code of: 2

GAPID Version: 1.4.0:developer
OS: linux amd64

Replaying trace from Raytracing demo from [SaschaWillems ](https://github.com/SaschaWillems/Vulkan)
API: Vulkan

Trace file:
[de.saschawillems.vulkanRaytracing_20190219_1328.gfxtrace.zip](https://github.com/google/gapid/files/2880172/de.saschawillems.vulkanRaytracing_20190219_1328.gfxtrace.zip)

Error in GAPID:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: Panic at command 2821:vkFlushMappedMemoryRanges(device: 3817529992, memoryRangeCount: 1, pMemoryRanges: 0xce6dee40) → VK_SUCCESS:
runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: 
--- *resolve.ContextListResolvable ---
 capture:<ID:<data:"\310\324\377\371\273\253\327\250lt\312\3458d\207\223\361\327r\001" > > config:<replay_device:<ID:<data:"\032\371\356T2\346\355=%\273\223\311:.]\217\301AO\335" > > > 
 Store():
  gapis/database/database.go:55  github.com/google/gapid/gapis/database.Build
  gapis/resolve/contexts.go:35  github.com/google/gapid/gapis/resolve.Contexts
  gapis/resolve/resolve.go:308  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:270  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:33  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:134  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:238  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:59  github.com/google/gapid/gapis/database.Build
  gapis/resolve/contexts.go:35  github.com/google/gapid/gapis/resolve.Contexts
  gapis/resolve/resolve.go:308  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:270  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:33  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:134  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:238  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

--- *resolve.GetResolvable ---
 path:<contexts:<capture:<ID:<data:"\310\324\377\371\273\253\327\250lt\312\3458d\207\223\361\327r\001" > > > > config:<replay_device:<ID:<data:"\032\371\356T2\346\355=%\273\223\311:.]\217\301AO\335" > > > 
 Store():
  gapis/database/database.go:55  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:26  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:388  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:197  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10548  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10550  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:982  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:1208  google.golang.org/grpc.(*Server).handleStream

 Build() #0:
  gapis/database/database.go:59  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:26  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:388  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:197  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10548  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10550  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:982  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:1208  google.golang.org/grpc.(*Server).handleStream
 [recovered]
	panic: 
--- *resolve.ContextListResolvable ---
 capture:<ID:<data:"\310\324\377\371\273\253\327\250lt\312\3458d\207\223\361\327r\001" > > config:<replay_device:<ID:<data:"\032\371\356T2\346\355=%\273\223\311:.]\217\301AO\335" > > > 
 Store():
  gapis/database/database.go:55  github.com/google/gapid/gapis/database.Build
  gapis/resolve/contexts.go:35  github.com/google/gapid/gapis/resolve.Contexts
  gapis/resolve/resolve.go:308  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:270  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:33  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:134  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:238  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

 Build() #0:
  gapis/database/database.go:59  github.com/google/gapid/gapis/database.Build
  gapis/resolve/contexts.go:35  github.com/google/gapid/gapis/resolve.Contexts
  gapis/resolve/resolve.go:308  github.com/google/gapid/gapis/resolve.ResolveInternal
  gapis/resolve/resolve.go:270  github.com/google/gapid/gapis/resolve.ResolveService
  gapis/resolve/get.go:33  github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve
  gapis/database/memory.go:134  github.com/google/gapid/gapis/database.(*record).resolve
  gapis/database/memory.go:238  github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1
  core/app/crash/crash.go:65  github.com/google/gapid/core/app/crash.Go.func1

--- *resolve.GetResolvable ---
 path:<contexts:<capture:<ID:<data:"\310\324\377\371\273\253\327\250lt\312\3458d\207\223\361\327r\001" > > > > config:<replay_device:<ID:<data:"\032\371\356T2\346\355=%\273\223\311:.]\217\301AO\335" > > > 
 Store():
  gapis/database/database.go:55  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:26  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:388  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:197  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10548  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10550  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:982  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:1208  google.golang.org/grpc.(*Server).handleStream

 Build() #0:
  gapis/database/database.go:59  github.com/google/gapid/gapis/database.Build
  gapis/resolve/get.go:26  github.com/google/gapid/gapis/resolve.Get
  gapis/server/server.go:388  github.com/google/gapid/gapis/server.(*server).Get
  gapis/server/grpc.go:197  github.com/google/gapid/gapis/server.(*grpcServer).Get
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10548  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
  core/app/auth/auth.go:110  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
  bazel-out/k8-opt/bin/gapis/service/linux_amd64_stripped/service_go_proto%/github.com/google/gapid/gapis/service/service.pb.go:10550  github.com/google/gapid/gapis/service._Gapid_Get_Handler
  external/org_golang_google_grpc/server.go:982  google.golang.org/grpc.(*Server).processUnaryRPC
  external/org_golang_google_grpc/server.go:1208  google.golang.org/grpc.(*Server).handleStream

[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x272a097]

goroutine 1235 [running]:
github.com/google/gapid/core/app/crash.Crash(0x46c1340, 0xc000aec260)
	core/app/crash/crash.go:89 +0xa7
github.com/google/gapid/core/app/crash.handler()
	core/app/crash/crash.go:56 +0x52
panic(0x46c1340, 0xc000aec260)
	GOROOT/src/runtime/panic.go:513 +0x1b9
github.com/google/gapid/gapis/database.(*memory).resolvePanicHandler(0xc000448480, 0x58cab40, 0xc000cf2030)
	gapis/database/debug.go:106 +0x7af
panic(0x4709da0, 0xc0018f2010)
	GOROOT/src/runtime/panic.go:513 +0x1b9
github.com/google/gapid/gapis/api.ForeachCmd.func1(0xc00004dac0, 0xc00004daf0)
	gapis/api/cmd_foreach.go:43 +0x152
panic(0x4862560, 0xa35a5f0)
	GOROOT/src/runtime/panic.go:513 +0x1b9
github.com/google/gapid/gapis/api/vulkan.DeviceMemoryObjectʳ.MappedLocationʷ(0x0, 0x0, 0x58cab40, 0xc000cbca50, 0x0, 0x0, 0x1, 0x1)
	bazel-out/k8-opt/genfiles/gapis/api/vulkan/api.go:52975 +0x37
github.com/google/gapid/gapis/api/vulkan.(*VkFlushMappedMemoryRanges).Mutate(0xc002424600, 0x58cab40, 0xc000cbca50, 0xb05, 0xc000aa31f0, 0x0, 0x0, 0x0, 0x0, 0x0)
	bazel-out/k8-opt/genfiles/gapis/api/vulkan/mutate.go:12265 +0x9d6
github.com/google/gapid/gapis/resolve.(*ContextListResolvable).Resolve.func1(0x58cab40, 0xc000cbca50, 0xb05, 0x590ed00, 0xc002424600, 0x0, 0x0)
	gapis/resolve/contexts.go:104 +0xad
github.com/google/gapid/gapis/api.ForeachCmd(0x58cab40, 0xc000cbc210, 0xc0018a0000, 0xb06, 0xe00, 0xc00004ddc8, 0x0, 0x0)
	gapis/api/cmd_foreach.go:54 +0x270
github.com/google/gapid/gapis/resolve.(*ContextListResolvable).Resolve(0xc0009d9e90, 0x58cab40, 0xc000cf2420, 0xc000cf21b0, 0xc0003b5f18, 0x2, 0x2)
	gapis/resolve/contexts.go:103 +0x1fa
github.com/google/gapid/gapis/database.(*record).resolve(0xc000cc2a80, 0x58cab40, 0xc000cf21b0, 0x0, 0x0)
	gapis/database/memory.go:134 +0x228
github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1()
	gapis/database/memory.go:238 +0xcc
github.com/google/gapid/core/app/crash.Go.func1(0xc0008c3e60)
	core/app/crash/crash.go:65 +0x43
created by github.com/google/gapid/core/app/crash.Go
	core/app/crash/crash.go:63 +0x3f
```

No output on logcat.

--- a/gapis/api/vulkan/api/memory.api
+++ b/gapis/api/vulkan/api/memory.api
@@ -211,7 +211,7 @@ cmd VkResult vkFlushMappedMemoryRanges(
       }
     }
 
-    if !(flushRange.memory in DeviceMemories) { vkErrorInvalidDeviceMemory(flushRange.memory) }
+    if !(flushRange.memory in DeviceMemories) { vkErrorInvalidDeviceMemory(flushRange.memory) } else {
     memoryObject := DeviceMemories[flushRange.memory]
     mappedLocation := as!u8*(memoryObject.MappedLocation)
     flushStart := flushRange.offset - memoryObject.MappedOffset
@@ -228,7 +228,7 @@ cmd VkResult vkFlushMappedMemoryRanges(
         // copy() contains an implicit read observation
         copy(memoryObject.Data[flushRange.offset:flushRange.offset + flushRange.size], (mappedLocation)[flushStart:flushStart + flushRange.size])
       }
-    }
+    }}
   }
   return ?
 }

